renderdoc/serialise/serialiser.cpp

```diff
@@ -334,13 +345,20 @@ uint32_t Serialiser<SerialiserMode::Writing>::BeginChunk(uint32_t chunkID, uint3
       {
         // write length, assuming it is an upper bound
         m_ChunkFixup = 0;
-        m_Write->Write(byteLength);
+        uint32_t chunkSizeLo = uint32_t(byteLength & 0xFFFFFFFF);
+        uint32_t chunkSizeHi = uint32_t((byteLength >> 32) & 0xFFFFFFFF);
+        RDCASSERT(chunkSizeHi == 0 || (c & Chunk64BitSize) != 0);
+        m_Write->Write(chunkSizeLo);
+        if(c & Chunk64BitSize)
+          m_Write->Write(chunkSizeHi);
```

Comment:
I'd rather check the flag and either write a 64-bit value or a 32-bit value, matching the logic in the read path. That way we don't make any unwanted endian-ness assumptions (even though I highly doubt that will ever change... you never know :smile: ).

Reply:
Agreed